docs/units/type_hints.rst

```diff
@@ -1,3 +1,11 @@
+Type Annotations Module
+***********************
+
+.. automodule:: astropy.units.typing
```

Comment:
Shouldn't this be at the end? (readthedocs seems stalled on uploading, so I cannot easily see how it looks).

Reply:
Done.